pages/docs/reference/js-modules.md

```diff
@@ -253,10 +250,53 @@ external fun sayHello(name: String)
 
 </div>
 
+### Automatic generation of external declarations with Dukat
+> Dukat is still experimental. If you encounter any problems, please report them in Dukat's [issue tracker](https://github.com/kotlin/dukat/issues).
+{:.note}
+
+[Dukat](https://github.com/kotlin/dukat) is a tool currently in development which allows the automatic conversion of TypeScript declaration files (`.d.ts`) into Kotlin external declarations. This aims to makes it more comfortable to use libraries from the JavaScript ecosystem in a type-safe manner in Kotlin, reducing the need for manually writing external declarations and wrappers for JS libraries.
+
+The Kotlin/JS Gradle plugin provides an integration with Dukat. When enabled, type-safe Kotlin wrappers are automatically generated for npm dependencies that either provide their own TypeScript definitions, or have definitions available in the [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) repository. You have two different ways of selecting if and when Dukat should generate declarations: at build time, and manually via a Gradle task.
```

Comment:
Actually we don't take d.ts files from DefinitelyTyped repo inside Gradle.
I think right now we get them only from a package based on package.json. cc @ilgonmic

Reply:
Well, we don't get files from DefinitelyTyped implicitly, but it can be done explicitly via NPM dependency, and it will work
`implementation(npm("@types/node", "*", true))` will provide Node.JS' TypeScript definitions to Dukat

BTW
> Kotlin wrappers

Do we use `Kotlin externals` for this?

Reply:
Paging @Schahen because I remember him talking about this functionality at some point, for some extra clarification.

Reply:
Right, we prefer using "external declaration".

Reply:
@ilgonmic I meant that right now we don't take **automatically** d.ts files anywhere else except package itself, even from `@types` scope. 

We had a plan to use `@types` scope as a fall back when a package doesn't have typing, but it's not implemented yet.

Reply:
I think there is no sense to find typings in DefinitelyTyped since all declarations from it are automatically published to `@types` scope.

Reply:
Yes, you are right
I just wrote it for information, that it can be covered by documentation

Reply:
There's sort of miscommunication, current scheme is following:

1. DefintelyTyped is published as a separate artifacts (under the "@types" prefix) to the npm registry
2. We resolve just `@types/`-prefixed npm dependencies.

that said - apart from tests we nowhere deal with DefinitelyTyped directly.

Reply:
I have adjusted the wording (on the new dukat-only page) and included a paragraph that specifies:

> If the repository of the dependency you wish to use does not provide TypeScript definitions, you can also use types provided via the [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) repository. In this case, make sure you add `npm` dependencies for both `your-package` and `@types/your-package` (with `generateExternals = true`).

Thanks for the clarification, @Schahen!